.travis.yml

```diff
@@ -48,6 +48,18 @@ script:
   - make DESTDIR=`pwd`/deploy install
   - popd
 
+  - cd "${TRAVIS_BUILD_DIR}/doc"
+  - doxygen
+  - touch html/.nojekyll
+
+deploy:
+  provider: pages
+  local-dir: doc/html
+  skip-cleanup: true
+  github-token: $GITHUB_TOKEN
+  on:
+    branch: master
```

Comment:
I would not push to master and add generated files to the repository. The master branch is the one everybody checks out and works with. I would rather add another branch (like gh-pages) or simply create a second project under prometheus-cpp which only stores the generated files (like in my example).